docs/DeveloperGuide.md

```diff
@@ -18,12 +44,162 @@
 
 {Describe the value proposition: what problem does it solve?}
```

Comment:
As with the other unfilled sections, it may be a good idea to draft out some points so that it is easier to form a proper paragraph by the time this is due.